src/Twig/Extension/BreadcrumbsExtension.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Twig\Extension;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Admin\BreadcrumbsBuilderInterface;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class BreadcrumbsExtension extends AbstractExtension
+{
+    /**
+     * @var BreadcrumbsBuilderInterface
+     */
+    private $breadcrumbsBuilder;
+
+    /**
+     * @internal
```

Comment:
why dont you place this on the class?

Reply:
This class being a Twig extension it is supposed to be used through filters, functions, etc. In this case I think we should guarantee BC for `renderBreadcrumbs` and `renderBreadcrumbsForTitle` methods, but the user is not supposed to instantiate this class.